### PR TITLE
Improve UART/DMA on stm32l4-multi

### PR DESCRIPTION
**Pull request by xvuko**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by agkaminski — APPROVED** [review]
LGTM

**Review by anglov — COMMENTED** [review]
I look over (not testing it) and looks sane. One MISRA comment

File: multi/stm32l4-multi/libmulti/include/libmulti/libdma.h [hunk 1]
```diff
@@ -28,7 +28,7 @@ enum { dma_per2mem = 0, dma_mem2per };
 enum { dma_spi = 0, dma_uart };
 
 
-enum { dma_ht = 0, dma_tc };
+enum { dma_ht = (1 << 0), dma_tc = (1 << 1) };
```
[the remark this answers is not in the document]
Comment (agkaminski) [reply multi/stm32l4-multi/libmulti/include/libmulti/libdma.h:31]: That's what I thought until recently, it turned out I misunderstood MISRA in this regard - casting enum -> int is legal

**Review by anglov — APPROVED** [review]

**Comment by Darchiv** [comment]
Please at least specify what type of changes this PR is

**Review by Darchiv — APPROVED** [review]

**Pull request merged**